File: implementation/js/math/binary_power.js
function binary_power(base, exponent){
    let res = 1;
    stack = [];
    while (exponent != 0){
        if (exponent%2 == 0){
            stack.push(2);
            exponent /= 2;
        } else{
            stack.push(1);
            exponent -= 1;
        }
    }
    for (let i=stack.length-1; i>=0; i--){
        res *= stack[i] == 1 ? base : res;
    }
    return res;
}

File: implementation/python/math/binary_power.py
def binary_power(base,expo):
    res = 1;
    stack = [];
    while expo != 0:
        if expo%2 == 0:
            stack.append(2);
            expo /= 2;
        else:
            stack.append(1);
            expo -= 1;
    print(stack[::-1])
    for e in stack[::-1]:
        res *= base if e == 1 else res;
        print(res)
    return res;